src/client/Microsoft.Identity.Client/ApiConfig/Executors/ManagedIdentityExecutor.cs

```diff
@@ -40,10 +40,20 @@ public ManagedIdentityExecutor(IServiceBundle serviceBundle, ManagedIdentityAppl
                 requestContext,
                 _managedIdentityApplication.AppTokenCacheInternal).ConfigureAwait(false);
 
-            var handler = new ManagedIdentityAuthRequest(
-                ServiceBundle,
-                requestParams,
+            // MSI factory logic - decide if we need to use the legacy or the new MSI flow
+
```

Comment:
We can have the "Executor" act like the factory. The executor is anyway responsible for creating a "handler", i.e. a request.  The SLCRequest has separate logic, because caching will be taken care of by CCA.